### PR TITLE
Tidy up package a bit

### LINKED ISSUE #102: Bump swift-crypto version range
Synchronize mysql-kit version dependencies
> .package(url: "https://github.com/apple/swift-crypto.git", "1.0.0" ..< "3.0.0"),
**to**
> .package(url: "https://github.com/apple/swift-crypto.git", "1.0.0" ..< "4.0.0"),

### PR DESCRIPTION
**Pull request by gwynne**
Fixes #102.

Also does some rather overdue general cleanup of the repo. No functional changes.

### CODE REVIEW

**Review by 0xTim — APPROVED** [review]

**Review by MahdiBM — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -22,16 +22,64 @@ env:
   MYSQL_HOSTNAME_B: 'mysql-b'
 
 jobs:
+  dependency-graph:
+    if: ${{ github.event_name == 'push' }}
+    runs-on: ubuntu-latest
+    container: swift:jammy
```
Comment (MahdiBM) [inline .github/workflows/test.yml:28]: Isn't this a tiny bit of inconsistency? 🤔
I think we usually explicitly specify the Swift version.

Comment (0xTim) [reply .github/workflows/test.yml:28]: Probably not for the dependency submission graph, it just needs to build it.

That said, I feel like the dependency graph stuff could be integrated directly into the main Vapor test stuff if it isn't already

Comment (gwynne) [reply .github/workflows/test.yml:28]: It is. This package just has the "needs extra services and the main CI doesn't have that capability quite yet" problem like most of Fluent's repos.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -22,16 +22,64 @@ env:
...
+    permissions:
+      contents: write
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+      - name: Fix Git configuration
+        run: |
+          git config --global --add safe.directory "${GITHUB_WORKSPACE}"
```
[the remark this answers is not in the document]
Comment (gwynne) [reply .github/workflows/test.yml:36]: They are, as it happens. Believe me, I didn't toss them in just for the fun of it, and I'm well aware the checkout action is _supposed_ to have dealt with it.

File: .github/workflows/test.yml (hunk 1 above)
Comment (gwynne) [reply .github/workflows/test.yml:28]: And I stopped explicitly specifying the Swift version for certain ancillary CI jobs (like this one) in the main CI long ago.

**Pull request merged**